modin/engines/base/frame/partition_manager.py

```diff
@@ -526,11 +526,19 @@ def apply_func_to_select_indices_along_full_axis(
         # Since we might be keeping the remaining blocks that are not modified,
         # we have to also keep the block_partitions object in the correct
         # direction (transpose for columns).
+        if not keep_remaining:
```

Comment:
Why was this change necessary? The condition here is handled below.

Reply:
It's not exactly the same condition, as you said above:
>the column partition is still shuffled together to the same worker, which is some significant work

so that changes are to do not creating column partitions for that partitions which will not be considered at computations.
Also, I've got these results after/before this changes:
```
45x500000, 112 cores

With fix:
Modin (1 value): 0.0579
Modin (all values): 0.9072

Without fix:
Modin (1 value):  0.2299
Modin (all values): 0.8566
```